logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/SendEditTextMessageUseCase.kt

```diff
@@ -0,0 +1,121 @@
+import com.wire.kalium.logic.kaliumLogger
+import com.wire.kalium.persistence.dao.message.MessageEntity
+import com.wire.kalium.util.DateTimeUtil
+import com.wire.kalium.util.KaliumDispatcher
+import com.wire.kalium.util.KaliumDispatcherImpl
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.withContext
+
+@Suppress("LongParameterList")
+
+/**
+ * Edits a text message
+ *
+ * @sample samples.logic.MessageUseCases.sendingEditBasicTextMessage
+ */
+class SendEditTextMessageUseCase internal constructor(
+    private val messageRepository: MessageRepository,
+    private val selfUserId: QualifiedID,
+    private val provideClientId: CurrentClientIdProvider,
+    private val slowSyncRepository: SlowSyncRepository,
+    private val messageSender: MessageSender,
+    private val dispatchers: KaliumDispatcher = KaliumDispatcherImpl
+) {
+
+    /**
+     * Operation to edit a text message.
+     *
+     * @param conversationId the id of the conversation the message belongs to
+     * @param originalMessageId the id of the message to edit
+     * @param text the edited content of the message
+     * @param mentions the edited mentions in the message
+     * @return [Either] [CoreFailure] or [Unit] //fixme: we should not return [Either]
+     */
+    suspend operator fun invoke(
+        conversationId: ConversationId,
+        originalMessageId: String,
+        text: String,
+        mentions: List<MessageMention> = emptyList(),
+    ): Either<CoreFailure, Unit> = withContext(dispatchers.io) {
+        slowSyncRepository.slowSyncStatus.first {
+            it is SlowSyncStatus.Complete
+        }
+        val generatedMessageUuid = uuid4().toString()
+
+        provideClientId().flatMap { clientId ->
+            val content = MessageContent.TextEdited(
+                editMessageId = originalMessageId,
+                newContent = text,
+                newMentions = mentions
+            )
+            val message = Message.Signaling(
+                id = generatedMessageUuid,
+                content = content,
+                conversationId = conversationId,
+                date = DateTimeUtil.currentIsoDateTimeString(),
+                senderUserId = selfUserId,
+                senderClientId = clientId,
+                status = Message.Status.PENDING
+            )
+            messageRepository.updateTextMessage(
+                conversationId = message.conversationId,
+                messageContent = content,
+                newMessageId = generatedMessageUuid,
+                editTimeStamp = message.date
+            ).flatMap {
+                    messageRepository.updateMessageStatus(
+                        messageStatus = MessageEntity.Status.PENDING,
+                        conversationId = message.conversationId,
+                        messageUuid = generatedMessageUuid
+                    )
+                }.map { message }
+        }.flatMap { message ->
+            messageSender.sendMessage(message)
```

Comment:
this part is a bit weird to me, why are we exactly doing map on updateMessageStatus() ? would it be enough to do not do map{ message } and just keep the rest as it is ?

Reply:
`updateMessageStatus` returns `Unit` when succeeds, but we need to pass a `message` to send it, that's why `Unit` is mapped to `message`, but you're right and `sendMessage` can be moved up a bit to have a direct access to `message` - I changed that.